include/mata/afa.hh

```diff
@@ -52,24 +61,88 @@ using Word = Mata::Nfa::Word;
 
 using StringDict = Mata::Nfa::StringDict;
 
+using StateSet = OrdVec<State>;
+
 using Alphabet = Mata::Nfa::Alphabet;
 
-/// A transition
+/*
+* A transition consists of a source state, a symbol and a vector of vector of states / vector of nodes
+* This corresponds to the formula in DNF
+*/
 struct Trans
 {
-	State src;
-  std::string formula;
+    State src; // source state
+    Symbol symb; // transition symbol
+    Nodes dst; // a vector of vectors of states
 
-	Trans() : src(), formula() { }
-	Trans(State src, const std::string& formula) : src(src), formula(formula) { }
+	Trans() : src(), symb(), dst() { }
+  	Trans(State src, Symbol symb, Node dst) : src(src), symb(symb), dst(Nodes(dst)) { }
+	Trans(State src, Symbol symb, Nodes dst) : src(src), symb(symb), dst(dst) { }
 
 	bool operator==(const Trans& rhs) const
 	{ // {{{
-		return src == rhs.src && formula == rhs.formula;
+		return src == rhs.src && symb == rhs.symb && dst == rhs.dst ;
 	} // operator== }}}
 	bool operator!=(const Trans& rhs) const { return !this->operator==(rhs); }
+
+    /*friend std::ostream& operator<<(std::ostream& os, const Trans trans)
+    {
+        return os << std::to_string(trans.src) + ":" + std::to_string(trans.symb) + "->" + std::to_string(trans.dst) + "\n";
+    }*/
+
 };
 
+using TransList = std::vector<Trans>;
+using TransRelation = std::vector<TransList>;
+
+/*
+* A tuple (result_nodes, sharing_list). If 'a' is an element of result_nodes, it means that there exists a symbol
+* 'b' such that sharing_list belongs to delta(a, b), where sharing_list is a node (vector of states). 
+*/
```

Comment:
The comments such this are very technical. It basically describes implementation but I would appreciate generally describe what InvResults are and then describe how this particular data structure represents it.

Reply:
I will rewrite comments later.